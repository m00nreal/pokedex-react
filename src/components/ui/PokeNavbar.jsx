import React, {useState} from 'react';
import {
    Collapse, Nav,
    Navbar,
    NavbarToggler,
    NavLink as BLink
} from "reactstrap";
import { Link, NavLink } from 'react-router-dom';



const PokeNavbar = () => {

    const [isOpen, setIsOpen] = useState(false);

    const toggle = () => {
        setIsOpen(!isOpen);
    };

    return (
        <Navbar color="danger" sticky="true" expand="md">
            <Link className="navbar-brand d-flex" to="/">
                <img className="brand" src={`${process.env.PUBLIC_URL}/assets/images/pkdx.png`} alt="Pokedex"/>
            </Link>
            <NavbarToggler onClick={toggle} className="navbar-dark"/>
            <Collapse isOpen={isOpen} navbar>
                <Nav className="mr-auto" navbar>
                    <NavLink
                        activeClassName="active"
                        to="/buscar"
                        className="nav-item nav-link"
                    >
                        Buscar
                    </NavLink>
                </Nav>
                <BLink
                    href="https://www.github.com/m00nreal"
                    target="_blank"
                    className="nav-item nav-link"
                >
                    <img className="d-none d-md-block" src={`${process.env.PUBLIC_URL}/assets/images/github.png`} alt="Github"/>
                    <span className="d-md-none">GitHub</span>
                </BLink>
            </Collapse>
        </Navbar>
    );
};

export default PokeNavbar;
