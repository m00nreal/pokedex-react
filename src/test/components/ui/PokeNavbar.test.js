import PokeNavbar from "../../../components/ui/PokeNavbar";
import React from "react";
import { shallow } from "enzyme";

describe('Tests in <PokeNavbar/>', function () {

    it('should match snapshot', function () {

        const wrapper = shallow(<PokeNavbar/>)
        expect(wrapper).toMatchSnapshot();
    });
});