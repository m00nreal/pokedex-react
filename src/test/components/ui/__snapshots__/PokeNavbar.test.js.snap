// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PokeNavbar/> should match snapshot 1`] = `
<Navbar
  color="danger"
  expand="md"
  sticky="true"
  tag="nav"
>
  <Link
    className="navbar-brand d-flex"
    to="/"
  >
    <img
      alt="Pokedex"
      className="brand flex-shrink-0"
      src="/pokedex-react/assets/images/pkdx.png"
    />
  </Link>
  <NavbarToggler
    className="navbar-dark"
    onClick={[Function]}
    tag="button"
    type="button"
  />
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={false}
    mountOnEnter={false}
    navbar={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <Nav
      className="mr-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        to="/buscar"
      >
        Search
      </NavLink>
    </Nav>
    <NavLink
      className="nav-item nav-link"
      href="https://github.com/m00nreal/pokedex-react"
      tag="a"
      target="_blank"
    >
      <img
        alt="Github"
        className="d-none d-md-block"
        src="/pokedex-react/assets/images/github.png"
      />
      <span
        className="d-md-none"
      >
        GitHub
      </span>
    </NavLink>
  </Collapse>
</Navbar>
`;
